core/modules/block/block.module

```diff
@@ -219,6 +261,24 @@ function block_custom_block_form($edit = array(), $stand_alone = TRUE) {
     '#weight' => -17,
   );
 
+  // Add language select dropdown if edit info is empty.
+  // This combination should only be set if adding a new block.
+  if (isset($edit['default_language']) && empty($edit['info'])) {
+    $options = array(LANGUAGE_NONE => t('- None -'));
+    $language_list = language_list();
+    foreach ($language_list as $value) {
+      $options[$value->langcode] = $value->name;
+    }
+
+    $form['default_language'] = array(
```

Comment:
Renaming this to `default_langcode` would help identify this value as being a string, rather than a language object.